api/v1alpha1/release_types.go

```diff
@@ -232,37 +261,105 @@ func (r *Release) IsValid() bool {
 	return meta.IsStatusConditionTrue(r.Status.Conditions, validatedConditionType.String())
 }
 
-// MarkProcessed marks the Release as processed.
+// MarkProcessed marks the Release overall as processed.
 func (r *Release) MarkProcessed() {
 	if !r.IsProcessing() || r.HasProcessingFinished() {
 		return
 	}
 
-	r.Status.Processing.CompletionTime = &metav1.Time{Time: time.Now()}
+	r.Status.ProcessingCompletionTime = &metav1.Time{Time: time.Now()}
 	conditions.SetCondition(&r.Status.Conditions, processedConditionType, metav1.ConditionTrue, SucceededReason)
 
 	go metrics.RegisterCompletedReleaseProcessing(
+		r.Status.ProcessingStartTime,
+		r.Status.ProcessingCompletionTime,
+		SucceededReason.String(),
+		r.Status.Target,
+	)
+}
+
+// MarkTenantProcessed marks the Release tenant portion as processed.
+func (r *Release) MarkTenantProcessed() {
+	if r.HasTenantProcessingFinished() {
+		return
+	}
+
+	r.Status.TenantProcessing.CompletionTime = &metav1.Time{Time: time.Now()}
+
+	go metrics.RegisterCompletedReleaseTenantProcessing(
 		r.Status.Processing.StartTime,
 		r.Status.Processing.CompletionTime,
 		SucceededReason.String(),
 		r.Status.Target,
 	)
 }
 
-// MarkProcessing marks the Release as processing.
+// MarkManagedProcessed marks the Release managed portion as processed.
+func (r *Release) MarkManagedProcessed() {
+	if r.HasManagedProcessingFinished() {
+		return
+	}
+
+	r.Status.ManagedProcessing.CompletionTime = &metav1.Time{Time: time.Now()}
+
+	go metrics.RegisterCompletedReleaseManagedProcessing(
+		r.Status.Processing.StartTime,
+		r.Status.Processing.CompletionTime,
+		SucceededReason.String(),
+		r.Status.Target,
+	)
+}
+
+// MarkProcessing marks the Release overall as processing.
 func (r *Release) MarkProcessing(message string) {
```

Comment:
I'd say that this have to go. HasProcessingFinished should be based in the state of each phase (HasTenantPipelineFinished, HasManagedPipelineFinished and HasFinalPipelineFinished).